### ISSUE TITLE
NTFS-formatted SD card cannot be mounted with `steamos-automount`

### ISSUE THREAD

**Issue by szescxz** [ublue-os/bazzite#522]
`systemctl status steamos-automount@mmcblk0p1.service` says:
```
Call failed: Mount option `uid=1000' is not allowed
```

**Comment by szescxz** [comment]
I believe I have figured out the cause.

In Fedora 38, the version of the [`udisks2`](https://packages.fedoraproject.org/pkgs/udisks2/udisks2/) package is same as the one used in SteamOS, `2.9.4`. In Fedora 39, the version is updated to `2.10.1`, which included a (breaking, IMO) change to the mount options: http://storaged.org/doc/udisks2-api/latest/mount_options.html#id-1.2.8.7
> The configurable mount options definition introduced in UDisks 2.9.0 considered the filesystem signature and corresponding filesystem driver an equal definition. Matching filesystem signature identifier was used as a filesystem type for mounting, as is common for most widely used filesystems.
>
> This has been changed in UDisks 2.10.0 and the configurable mount options gained ability to specify filesystem driver (i.e. a kernel or fuse driver) for a particular filesystem signature, along with a definition of filesystem driver priorities if necessary. In the following paragraphs the term 'filesystem signature' typically means 'filesystem type' unless stated otherwise.

In SteamOS Btrfs project, instead of letting `udisks2` select the NTFS driver automatically, [they specifically selected `lowntfs-3g` as the driver for NTFS filesystems](https://gitlab.com/popsulfr/steamos-btrfs/-/blob/462ad3c8133872582affe68f87f19605e2eb996c/files/usr/lib/hwsupport/steamos-automount.sh.patch.old.20231104103824.20231031.1000#L58).
With the 2.10.x update of `udisks2`, if continue using `lowntfs-3g`, [the line that sets the mount options](https://gitlab.com/popsulfr/steamos-btrfs/-/blob/462ad3c8133872582affe68f87f19605e2eb996c/files/usr/lib/hwsupport/steamos-automount.sh.patch.old.20231104103824.20231031.1000#L71) needs to be updated.

For the contents of `/etc/udisks2/mount_options.conf`, current behavior is:
```
[defaults]
lowntfs-3g_allow=uid=$UID,gid=$GID,umask,dmask,fmask,locale,norecover,ignore_case,windows_names,compression,nocompression,big_writes,nls,nohidden,sys_immutable,sparse,showmeta,prealloc,rw,noatime,lazytime,uid=1000,gid=1000,big_writes,umask=0022,ignore_case,windows_names
```
With `udisks2` 2.10.x, this will need to change to:
```
[defaults]
ntfs:lowntfs-3g_allow=uid=$UID,gid=$GID,umask,dmask,fmask,locale,norecover,ignore_case,windows_names,compression,nocompression,big_writes,nls,nohidden,sys_immutable,sparse,showmeta,prealloc,rw,noatime,lazytime,uid=1000,gid=1000,big_writes,umask=0022,ignore_case,windows_names
```
Basically, prefix `ntfs:` before `lowntfs-3g_allow=`.

I'm not sure about how to change this without affecting other filesystems, nor how to ensure backward compatibility with older version of `udisks2` (which Valve is probably going to stick with in the forseeable future).

**Comment by KyleGospo** [comment]
Should be fixed with: https://github.com/ublue-os/bazzite/commit/11408e14acf80c29175d6fb78d845581da3dbd4b

**Comment by szescxz** [comment]
> Should be fixed with: [11408e1](https://github.com/ublue-os/bazzite/commit/11408e14acf80c29175d6fb78d845581da3dbd4b)

Like I said, fixes like that will break backward compatibility. Tested the patch on F38:
```
$ sudo /usr/libexec/steamos-automount add mmcblk0p1
Call failed: Mount option `uid=1000' is not allowed
Running as unit: run-u180.service
Finished with result: success
Main processes terminated with: code=exited/status=0
Service runtime: 541ms
CPU time consumed: 541ms
Sent URL to steam: steam://system/devicemountresult/mmcblk0p1/1 (steam://system/devicemountresult/%6d%6d%63%62%6c%6b%30%70%31%2f%31)
Error mounting /dev/mmcblk0p1 (status = 1)
```
I think upstream Fedora is not going bump `udisks2` version to 2.10.x on F38 ever, so unless dropping support for F38 _now_ is intended for Bazzite, this patch will also need to include some kind of version checks as well.

**Comment by KyleGospo** [comment]
We no longer build F38 images, F39 only going forward.

**Comment by szescxz** [comment]
I see, just noticed that you've already removed F38 builds on GitHub Actions. Working on F39 so closing.

**Comment by szescxz** [comment]
Reopening again.
@KyleGospo Seems like this was not implemented in https://github.com/ublue-os/bazzite/commit/4be4e7a5ac0d4c6c263b2706223a889c54cc2d24#diff-5eb727e52d9c578ef2267af913d6ba1fa93ad1b1b2ca21e35ff4576dbae94596R58 and finally broken due to https://github.com/ublue-os/bazzite/commit/ac53c8bd6cedf78b431fb1a0bb2bfd38b0a08a9f#diff-6095ec0a2434e296d7ae243c339ed199a2d571fc488c01cdeecfdb59a22af28cL50?

**Comment by KyleGospo** [comment]
@szescxz thanks, fixed in next release

**Comment by szescxz** [comment]
https://github.com/ublue-os/bazzite/commit/8110691306470fd35ca02bb4170621ff543d5199
Rebased to testing and confirmed, thank you very much!
